RetroAchievements/RAWeb
# fix: sync i18n plurals correctly

i18next plural forms are not syncing correctly across locales with the `pnpm i18n:sync` command.

For example, while this is valid in en_US:
```json
{
    "supportedGameFilesCountLabel_one": "There is currently <1>{{count, number}}</1> supported game file hash registered for this game.",
    "supportedGameFilesCountLabel_other": "There are currently <1>{{count, number}}</1> supported game file hashes registered for this game.",
}
```

It is actually _invalid_ in pl_PL, as pl_PL has multiple plural forms:
```json
{
    "supportedGameFilesCountLabel_one": "Obecnie jest <1>{{count, number}}</1> zarejestrowany hash pliku gry",
    "supportedGameFilesCountLabel_few": "Obecnie są <1>{{count, number}}</1> zarejestrowane hashe pliku gry",
    "supportedGameFilesCountLabel_many": "Obecnie jest <1>{{count, number}}</1> zarejestrowanych hashy plików gry",
    "supportedGameFilesCountLabel_other": "Obecnie jest <1>{{count, number}}</1> zarejestrowanych hashy plików gry"
}
```

When there are 2 supported game files and the user is using pl_PL, i18next is naturally reaching for the "_few" key, but it currently isn't present, causing that string to fall back to English.

After the changes in this file, `pnpm i18n:sync` syncs plural keys correctly. The new sync script is intelligent enough to recognize what plural forms the target locale file requires, and it writes the keys accordingly.

---

**How to test this PR**
```
pnpm install
```

Add the following to en_US.json:
```
    "people_one": "Person",
    "people_other": "People"
```
Run the command:
```
pnpm i18n:sync
```
Proper plural forms should be written to each JSON file, ie: pl_PL:
```json
    "people_one": "Person",
    "people_few": "People",
    "people_many": "People",
    "people_other": "People"
```
